extension/libs/turndown.js

```diff
@@ -75,7 +75,47 @@ var TurndownService = (function () {
       tagNames.some(function (tagName) {
         return node.getElementsByTagName(tagName).length
       })
-    )
+    );
+  }
+
+  function inferAlt(img) {
+    // 1) Keep existing alt
+    const existing = img.getAttribute("alt");
+    if (existing && existing.trim()) {
+      return existing.trim();
+    }
+
+    // 2) Pull from a <figcaption> if inside a <figure>
+    const figcap = img.closest("figure")?.querySelector("figcaption");
+    if (figcap && figcap.textContent.trim()) {
+      return figcap.textContent.trim();
+    }
+
+    // 3) filename fallback & special “avatar” case
+    try {
+      const url = new URL(img.src, location.href);
+      const raw = url.pathname
+        .split("/")
+        .pop()
+        .replace(/\.\w+$/, "")
+        .replace(/[-_]+/g, " ")
+        .trim();
+
+      // if it mentions “avatar” in the URL path or host, treat it as an avatar
+      if (/avatar/i.test(url.pathname) || /avatar/i.test(url.hostname)) {
+        return "Avatar";
+      }
+
+      // skip pure numbers — otherwise use the humanized filename
+      if (raw && !/^\d+$/.test(raw)) {
+        return raw.charAt(0).toUpperCase() + raw.slice(1);
+      }
+    } catch (e) {
+      console.error(e);
```

Comment:
Catching and only logging errors might hide important issues. Consider whether we want to fail silently or provide more context about what failed. 

```suggestion
      console.warn('Failed to parse image URL for alt text inference:', img.src, e);
```